### PR TITLE
[AIP-140] Abbreviations

### LINKED ISSUE #167: AIP-140: Field names / abbreviations
Old linter rule: http://g3doc/google/api/tools/linter/g3doc/rules/abbreviations

### PR DESCRIPTION
**Pull request by lukesneeringer**
~Uses #207 _and_ #208 as a base.~ Those are merged.

Fixes #167.

### CODE REVIEW

**Comment by lukesneeringer** [comment]
Ignore my rebase failure -- squash and merge will make it not matter.

File: rules/aip0140/abbreviations.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	"fmt"
+	"strings"
+
+	"github.com/googleapis/api-linter/lint"
+	"github.com/jhump/protoreflect/desc"
+)
+
+// checkAbbreviations checks for each long name in the descriptor's name,
+// and returns problems if one is found that suggests the abbreviation instead.
+func checkAbbreviations(d desc.Descriptor, x func(string) string) (problems []lint.Problem) {
```
Comment (mingzhi) [inline rules/aip0140/abbreviations.go]: Consider to give `x` a more informative name, e.g., `formatCase` or a better name.

File: rules/aip0140/abbreviations.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	LintMessage: func(m *desc.MessageDescriptor) []lint.Problem {
+		return checkAbbreviations(m, strings.Title)
+	},
+}
+
+var abbreviationsEnum = &lint.EnumRule{
+	Name: lint.NewRuleName("core", "0140", "enum-names", "abbreviations"),
+	URI:  "https://aip.dev/140#abbreviations",
+	LintEnum: func(e *desc.EnumDescriptor) (problems []lint.Problem) {
+		problems = append(problems, checkAbbreviations(e, strings.Title)...)
```
Comment (mingzhi) [inline rules/aip0140/abbreviations.go]: Nitpicking: use `problems := checkAbbreviations(e, strings.Title)`.

File: rules/aip0140/abbreviations.go (hunk 1 above)
Comment (lukesneeringer) [reply rules/aip0140/abbreviations.go]: Done.

File: rules/aip0140/abbreviations.go (hunk 2 above)
Comment (lukesneeringer) [reply rules/aip0140/abbreviations.go]: Done.

**Pull request merged**